test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -1756,6 +1756,363 @@ describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
         expect(fooFunctionRole).to.equal(fooFunctionConfig.role);
       });
     });
+
+    it('should fail if `functions[].image` references image without path and uri', async () => {
```

Comment:
Aren't those tests duplicate of ones in `deployFunction.test.js` ?

Reply:
That's correct, just tested in different use case scenario - do you feel like these should be just moved to `provider.test.js`?

Reply:
I've moved these to `provider.test.js` from both `deployFunction.test.js` and `functions.test.js`